scalafix-core/src/main/scala/scalafix/internal/v0/LegacyCodePrinter.scala

```diff
@@ -30,21 +31,58 @@ class LegacyCodePrinter() {
       text.append(start)
       var first = true
       trees.foreach { tree =>
-        fn(tree)
         if (first) {
           first = false
         } else {
           text.append(", ")
         }
+        fn(tree)
       }
       text.append(end)
     }
   }
 
-  private def pprint(sig: s.Signature): Unit = sig match {
-    case s.ValueSignature(tpe) =>
-      pprint(tpe)
-    case _ =>
+  private def pprintTypeParameters(scope: Option[s.Scope]): Unit = {
+    scope.foreach { scope =>
+      if (scope.symbols.nonEmpty) {
+        mkString("[", scope.symbols, "]")(emit)
+        text.append(" => ")
+      }
+    }
+  }
+
+  private def pprint(signature: s.Signature): Unit = {
+    signature match {
+      case sig: s.ClassSignature =>
```

Comment:
@olafurpg should we support those? I gave it a try and it was not so easy, this is why I left a stub here.